ocean-client/src/programs/vault-maxi-program.ts

```diff
@@ -747,18 +819,22 @@ export class VaultMaxiProgram extends CommonProgram {
       if (!this.keepWalletClean) {
         token.amount = '' + BigNumber.min(token.amount, expectedA)
       }
-      if (this.negInterestWorkaround && interestA.lt(0)) {
-        token.amount = '' + interestA.times(1.005).plus(token.amount) //neg interest with the bug is implicitly added to the payback -> send in "wanted + negInterest"
+      if (this.isSingleMintB) {
+        collateralTokens.push(token)
+      } else {
+        if (this.negInterestWorkaround && interestA.lt(0)) {
+          token.amount = '' + interestA.times(1.005).plus(token.amount) //neg interest with the bug is implicitly added to the payback -> send in "wanted + negInterest"
```

Comment:
can we remove this special handling?
If I recall correctly, we added it due to the very first negative interest introduction bug, where calculating it exactly would lead to an issue

Reply:
yes